patches/server/0725-Add-BlockBreakBlockEvent.patch

```diff
@@ -0,0 +1,83 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Sun, 3 Jan 2021 17:58:11 -0800
+Subject: [PATCH] Add BlockBreakBlockEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/Block.java b/src/main/java/net/minecraft/world/level/block/Block.java
+index d6a3f3a2edae806b0ebf5bf5ac445116c0d64535..49d2e95f1c925cd3eaa850a7e6385eb54cc14aae 100644
+--- a/src/main/java/net/minecraft/world/level/block/Block.java
++++ b/src/main/java/net/minecraft/world/level/block/Block.java
+@@ -331,6 +331,19 @@ public class Block extends BlockBehaviour implements ItemLike {
+         }
+ 
+     }
++    // Paper start
++    public static boolean dropItemsWithEvent(BlockState blockState, LevelAccessor generatorAccess, BlockPos blockPosition, BlockPos source, @Nullable BlockEntity tileEntity) {
++        if (generatorAccess instanceof ServerLevel) {
++            io.papermc.paper.event.block.BlockBreakBlockEvent event = new io.papermc.paper.event.block.BlockBreakBlockEvent(org.bukkit.craftbukkit.block.CraftBlock.at(generatorAccess, blockPosition), org.bukkit.craftbukkit.block.CraftBlock.at(generatorAccess, source), Block.getDrops(blockState, generatorAccess.getMinecraftWorld(), blockPosition, tileEntity).stream().map(org.bukkit.craftbukkit.inventory.CraftItemStack::asBukkitCopy).collect(java.util.stream.Collectors.toList()));
++            event.callEvent();
++            event.getDrops().stream().map(org.bukkit.craftbukkit.inventory.CraftItemStack::asNMSCopy).forEach((itemstack) -> {
```

Comment:
Feels like a small for loop would be nicer here, as I reckon this is called for every block break?

Reply:
It's not, just when other blocks cause a block to be broken, like a piston extension or water flowing. BUT it doesnt hurt to replace the streams anyways.